src/type_utils.jl

```diff
@@ -0,0 +1,14 @@
+"replace `Symbol`s with `QuoteNode`s to avoid expression interpolation"
+wrap_symbol_quotenode(param) = param isa Symbol ? QuoteNode(param) : param
+
+"Construct the expression for qualifying a type with given parameters"
+function construct_type_ex(type, parameters)
```

Comment:
```suggestion
function construct_type_expr(type, parameters)
```
Just a personal preference for a naming convention.